x509/x509.go

```diff
@@ -628,6 +636,11 @@ type Certificate struct {
 
 	// CT
 	SignedCertificateTimestampList []*ct.SignedCertificateTimestamp
+
+	// Used to speed up the zlint checks. Populated by the GetParsedDNSNames method.
+	parsedDNSNames []ParsedDomainName
+	// Used to speed up the zlint checks. Populated by the GetParsedCommonName method
+	parsedCommonName *ParsedDomainName
```

Comment:
I'm a bit surprised that one of these is an array of objects and one is a pointer. Is there a reason not to just store both as objects on certificate?

Reply:
Both fields needs to be nillable. 
`parsedDNSNames` is a slice, because the DNS SANs are a slice. So storing parsed DNS SANs in a slice makes sense. 

For the CommonName, I need to discern between common name being an empty string and it not being parsed yet. This could be done by checking is parsedCommonName.ParsedDomain is nil or not, but in general using a pointer to the whole ParsedDomainName struct seems cleaner.

Reply:
Also, as the linters first try to parse the common name, and then proceed to SAN DNS names if parsing of the common name succeeds, it was a reasonable choice to split those two concerns into separate fields in the cert struct and separate function calls.